WcaOnRails/app/models/competition.rb

```diff
@@ -2405,6 +2405,28 @@ def set_form_data_series(form_data_series, current_user)
     self.competition_series = competition_series
   end
 
+  def payments_enabled?
+    competition_payment_integrations.exists?
+  end
+
+  def payment_account_for(integration_name)
+    validate_integration_name!(integration_name)
+
+    payment_integrations = competition_payment_integrations.where(
+      connected_account_type: CompetitionPaymentIntegration::AVAILABLE_INTEGRATIONS[integration_name],
+    )
+
+    return nil unless payment_integrations.count > 0
+
+    # Take the first result as `where` always returns an array
+    payment_integrations.first.connected_account
```

Comment:
```suggestion
    return competition_payment_integrations.first(
      connected_account_type: CompetitionPaymentIntegration::AVAILABLE_INTEGRATIONS[integration_name],
    )&.connected_account
```

Reply:
Context: In Rails ActiveRecord, `first` can act the same way `where` does, taking certain search criteria using keywords. It's basically a drop-in "return only the first one found instead of a whole array" replacement for `where`.

Also, big fat no on the extra `unless count > 0` because that fires an extra SQL query.
Nice to know for later, but definitely not necessary here: `Competition.where(foo: 'bar').exists?` uses an optimised query to check for existence on raw SQL level, without evaluating the full count.

Reply:
Flagging this as another instance where I would expect no SQL query given that we've already assigned the result of a `where` clause to `payment_integrations`

Reply:
How sure are you on that behaviour of `.first`? In the code, your suggestion is generating `TypeError: no implicit conversion of Integer into Hash`, and the [Rails docs](https://apidock.com/rails/v5.2.3/ActiveRecord/FinderMethods/first) don't list an example with that usage - they all seem to use .where.first

Reply:
My bad, it's actually `CoolModel.find_by(column: 'amazing_value')`

Reply:
Re query count: The CPI.where chaining does not fire any queries.

In your original version, the .count fires the first query, and the .first fires the second query. But the point is, you're not really using that count in a reasonable way: If there is nothing present, the `first` will yield nil and you can use that to react to the "there's nothing there" case

Reply:
Cool, thanks! Updated it to find_by :+1: